app/models/payment.rb

```diff
@@ -148,6 +152,31 @@ def retrieve_payment_intent
     Rails.logger.debug { "retrieve_payment_intent payment => #{inspect}}" }
   end
 
+  # https://docs.stripe.com/api/refunds
+  # reasons duplicate, fraudulent, or requested_by_customer
+  def refund_payment(reason: 'requested_by_customer')
+    return false unless received?
+
+    begin
+      Rails.logger.info "refund_payment: #{id} stripe_payment_id: #{stripe_payment_id} status: #{status}"
+      self.refund = Stripe::Refund.create({ payment_intent: stripe_payment_id,
+                                            reason:,
+                                            metadata:       { event_id:               ticket_request.event.id,
+                                                              event_name:             ticket_request.event.name,
+                                                              ticket_request_id:      ticket_request.id,
+                                                              ticket_request_user_id: ticket_request.user_id } })
+      self.stripe_refund_id = refund.id
+      self.status = STATUS_REFUNDED
+      Rails.logger.info { "refund_payment success stripe_refund_id [#{stripe_refund_id}] status [#{status}]" }
+      log_refund(refund)
+      save
```

Comment:
Might want to call `save!` which raises exception if it can't be saved. 

Then you could add additional rescue:

```ruby
   rescue StandardError => e
     Rails.logger.error("#{e.class.name} ERROR during a refund: #{e.message.colorize(:red)}")
```